### ISSUE TITLE
NaN in the jacobian for lepton collision

### ISSUE THREAD

**Issue by claudioseveri** [mg5amcnlo/mg5amcnlo#75]
Hello everyone,
phase space generation for lepton collisions occasionally gives NaN in the jacobian `xjac0`.

In `generate_ee_tau_y`, the jacobian is multiplied by a multichannel factor that contains
`(1d0-tau_born)**(1d0-2*get_ee_expo())`
If `1d0-tau_born` is zero within machine precision, which is possible in lepton collisions, the above is actually a division by zero. 

Since all checks for unphysical configurations look something like this,
`      ! filter unphysical configurations`
`      if (xjac0.lt.0d0) ... `
the NaN is not caught in later checks, and propagates down.

I would recommend handling the case tau_born = 1 in a cleaner way, either rejecting the PS configuration, or using a finite expression for the jacobian if it exists.

**Comment by marcozaro** [comment]
Ciao Claudio,
can you tell us more about what process you are lookling at?
This fix
```

diff ../../../Template/NLO/SubProcesses/genps_fks.f genps_fks.f
4284a4285
>       double precision omtau_born
4381a4383,4384
>         ! better numerical accuracy
>         omtau_born = omx_ee(1) + omx_ee(2) - omx_ee(1)*omx_ee(2)
4384,4385c4387,4388
<         if (bw_exists) xjac0 = xjac0 * (1d0-tau_born)**(1d0-2*get_ee_expo()) / 
<      $       ( 1d0/((tau_born-tau_m)**2 + tau_m*tau_w) + (1d0-tau_born)**(1d0-2*get_ee_expo()))
---
>         if (bw_exists) xjac0 = xjac0 * (omtau_born)**(1d0-2*get_ee_expo()) / 
>      $       ( 1d0/((tau_born-tau_m)**2 + tau_m*tau_w) + (omtau_born)**(1d0-2*get_ee_expo()))
```

should improve the numerical accuracy

**Comment by claudioseveri** [comment]
Dear Marco,
for future reference the process is top pair production with subsequent decay into four leptons, 
`generate e+ e- > t t~ > b b~ e+ ve mu- vm~ QCD=0 QED=6 [LOonly]`

You fix solves the problem. (One may add an explicit check for `omtau_born.eq.0d0` but this should be very very unlikely on a 64 bit machine.) 

Now that both `1-x` and `1-tau` are available, I suggest a similar correction to all other places in `generate_ee_tau_y` where `1-x` or `1-tau` are used.

Thanks
Claudio

**Comment by marcozaro** [comment]
Hi Caludio,
thanks. I have pushed the fix in 3.5.3 (aa00b5c22).
About the other occurrences of 1-tau and 1-x. They are not as bad as in that case, in the sense that it is crucial to keep track of 1-x only when you do the generation with the generate_x_ee function (ie. doing adaptive sampling towards x->1). For the rest, it should not be as bad as in this case.
Cheers,

Marco
